### ISSUE TITLE
Cant open GoldSrc games from Steam

### ISSUE THREAD

**Issue by shoober420** [ValveSoftware/halflife#3112]
For quite some time now, I havent been able to open any GoldSrc games on my Gentoo machine. I have to manually go to the Half-Life directory and run `./hl.sh -game foo` to launch different Half-Life mods. Any idea why this is happening?

System Information:
https://gist.github.com/shoober420/35af31e8e121a2a55a1004a37597ca60

**Comment by niobium93** [comment]
Same problem on Arch Linux. Game launches fine when manually running hl.sh, but the process just hangs when launched from the steam client, with no command line output at all. Disabling shader pre-caching works around the issue, but there's no way to do that per-game, and I'd rather not cripple all my other games...

**Comment by niobium93** [comment]
Figured out how do fix it without disabling shader pre-caching for everything. Paste this into your launch options:
```
env -u MESA_DISK_CACHE_SINGLE_FILE -u MESA_GLSL_CACHE_DIR %command%
```
Both of those environment variables need to be unset. Either one of them still breaks the game.

**Comment by Tynach** [comment]
@niobium93, your fix isn't working for me. I even tried renaming `MESA_GLSL_CACHE_DIR` to `MESA_SHADER_CACHE_DIR`, to no avail. I also can't run it from command line; the same thing happens if I run either `hl.sh` or `hl_linux` directly (of course adding the directory it's in to `LD_LIBRARY_PATH`). When I do this, I get the following on the command line:

```
Using breakpad crash handler
[S_API] SteamAPI_Init(): Loaded '/home/tynach/.local/share/Steam/linux32/steamclient.so' OK.
src/common/pipes.cpp (883) : fatal stalled cross-thread pipe.
src/common/pipes.cpp (883) : fatal stalled cross-thread pipe.
src/common/pipes.cpp (883) : Fatal assert; application exiting
src/common/pipes.cpp (883) : Fatal assert; application exiting
07/13 12:41:52 minidumps folder is set to /tmp/dumps
07/13 12:41:52 Init: Installing breakpad exception handler for appid(hl_linux)/version(1.0)/tid(21451)
assert_20240713124152_3.dmp[21575]: Uploading dump (out-of-process)
/tmp/dumps/assert_20240713124152_3.dmp
_ExitOnFatalAssert
assert_20240713124152_3.dmp[21575]: Finished uploading minidump (out-of-process): success = yes
assert_20240713124152_3.dmp[21575]: response: Discarded=1
assert_20240713124152_3.dmp[21575]: file ''/tmp/dumps/assert_20240713124152_3.dmp'', upload yes: ''Discarded=1''
```

The first two lines are the only output for a long time, and then after exactly 3 minutes it prints the rest and quits. Though, oddly enough, it quits early enough that the last 3 lines appear after my shell's prompt, so I removed that stray shell prompt at the start of the third-to-last line.

I'd share that `assert_20240713124152_3.dmp` file, but it has a LOT of info in it about my system (for example, the values of *every* environment variable that's set on my system). I'm fine with Valve having that information, but I don't want to put it up on Github publicly.

Publicly, I will say that I'm using two 1080p monitors (with the left monitor set up as the primary one, and the one I want the game to play on), have an AMD Radeon R9 290X, and am using KDE Neon (without having updated to the latest version of KDE (still on Plasma 5.27.10, but have kept mostly up to date on other system updates.. I say 'mostly' because I have just now noticed that some have come in since I last looked)). KDE Neon is based on Ubuntu 22.04.

I'm also using the 'Kisak' PPA of more up-to-date Mesa GPU drivers, and have already set my system up so that it's using the `amdgpu` kernel driver so that Vulkan can work. Other vulkan games work, and Proton's DXVK works as well.

**Comment by niobium93** [comment]
@Tynach that old fix hasn't been needed for a long time. Both the game and mesa have updated and whatever the problem was before is now long gone. You're probably having a completely different issue from me - I've never had those asserts myself. Open a new issue. Also Half-Life 1 doesn't use Vulkan, only OpenGL.

**Comment by Tynach** [comment]
@niobium93, I mentioned the stuff about Vulkan because of this thread on Steam:
https://steamcommunity.com/app/221410/discussions/0/3123786787136402921/?ctp=3#c3105773861176631493

That response I linked to has enough history in its quote chain to give you my reasons for saying what I did. I was basically just covering any responses to my post I might get that refer to that conversation, or refer to the same things that conversation refers to.

Has your hardware changed since the old fix was needed? Or anything else about your system?

Also, I've seen some people blame PulseAudio (found when I googled `src/common/pipes.cpp (883) : fatal stalled cross-thread pipe.`), but since that problem supposedly also stopped Steam itself from running, I don't think it's related.

**Comment by Tynach** [comment]
Yeah, nevermind everything I said. Something screwy was going on with Steam, and I noticed I couldn't launch *any* games. Restarting Steam helped only for a little bit, but after a reboot I was fine.

Thought for sure I said all that already, but I guess I just meant to and forgot to actually do so. But now you're caught up.
